truetype/truetype.go

```diff
@@ -306,6 +317,65 @@ func (f *Font) parseMaxp() error {
 	return nil
 }
 
+func (f *Font) parseName() error {
+	if len(f.name) < 18 {
+		return FormatError("name data too short")
+	}
+	if u16(f.name, 0) != 0 {
+		return FormatError(fmt.Sprintf("invalid format specifier: %d", u16(f.name, 0)))
+	}
+	numNames := int(u16(f.name, 2))
+	stringOffset := u16(f.name, 4)
+	font := &nr{}
+
+	for offset, i := 6, 0; i < numNames; i, offset = i+1, offset+12 {
+		nameID := u16(f.name, offset+6)
+		switch nameID {
+		case 1, 2:
+			/* Keeping these here in case they're requested later.
+
+			platformSpecificID, languageID,platformID := u16(f.name, offset+2), u16(f.name, offset+4),u16(f.name, offset)
+			*/
+			length, nameOffset := u16(f.name, offset+8), u16(f.name, offset+10)
+
+			// Assume a non-UTF-16 value
+			value := f.name[stringOffset+nameOffset : stringOffset+nameOffset+length]
+
+			// but just in case... if the first byte is 0, as in luxi's 13th nameRecord,
+			// assume UTF-16, try to parse it if its length is even
+			if value[0] == 0 && length&1 == 0 {
+				var err error
+
+				enc := unicode.UTF16(unicode.BigEndian, unicode.IgnoreBOM)
+				value, err = ioutil.ReadAll(transform.NewReader(bytes.NewReader(value), enc.NewDecoder()))
+				if err != nil {
+					return err
+				}
+			}
+			// TODO: find a fail-safe method for determine if a byte slice or string is
+			// UTF-16 or not.
+			font.fontInfo[nameID-1] = string(value)
+		}
+
+		if font.fontInfo[0] != "" && font.fontInfo[1] != "" {
+			f.nameRecord = font
+			return nil
+		}
+	}
+
+	return nil
+}
+
+// Name returns the TrueType font's name
```

Comment:
Drop the "TrueType". We're already in "package truetype", and don't deal with any other sort of font files.